torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,167 @@
+import io
+import pathlib
+import re
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Tuple, Iterator, Iterable, TypeVar, Union
+
+import numpy as np
+import torch
+from torchdata.datapipes.iter import (
+    IterDataPipe,
+    Demultiplexer,
+    Mapper,
+    Shuffler,
+    Filter,
+    IterKeyZipper,
+    ZipArchiveReader,
+)
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+    DatasetType,
+)
+from torchvision.prototype.datasets.utils._internal import INFINITE_BUFFER_SIZE
+
+T = TypeVar("T")
+
+FILE_NAME_PATTERN = re.compile(r"(frame|image)_(?P<idx>\d+)[.](flo|png)")
+
+try:
+    from itertools import pairwise  # type: ignore
+except ImportError:
+    from itertools import tee
+
+    def pairwise(iterable: Iterable[T]) -> Iterable[Tuple[T, T]]:
+        a, b = tee(iterable)
+        next(b, None)
+        return zip(a, b)
+
+
+class IntCategoryGrouper(IterDataPipe[Tuple[Tuple[str, T], Tuple[str, T]]]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, T]]) -> None:
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[Tuple[Tuple[str, Any], Tuple[str, Any]]]:
+        for item1, item2 in pairwise(sorted(self.datapipe)):
+            if pathlib.Path(item1[0]).parent != pathlib.Path(item2[0]).parent:
+                continue
+
+            yield item1, item2
+
+
+class SINTEL(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "sintel",
+            type=DatasetType.IMAGE,
+            homepage="",
+            valid_options=dict(
+                split=("train", "test"),
+                pass_=("clean", "final"),
+            ),
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        archive = HttpResource(
+            "http://sintel.cs.washington.edu/MPI-Sintel-complete.zip",
+            sha256="",
+        )
+        return [archive]
+
+    def _classify_train_test(self, data: Tuple[str, Any], *, config: DatasetConfig) -> bool:
+        path = pathlib.Path(data[0])
+        return config.split in str(path.parent)
+
+    def _classify_archive(self, data: Tuple[str, Any], *, config: DatasetConfig) -> Union[int, None]:
```

Comment:
Both are equivalent, but `Optional` is the preferred way to go.

```suggestion
    def _classify_archive(self, data: Tuple[str, Any], *, config: DatasetConfig) -> Optional[int]:
```